pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -608,6 +599,18 @@ func (npm *nominatedPodMap) AddNominatedPod(pod *v1.Pod, nodeName string) {
 	npm.Unlock()
 }
 
+// NominatedPodsForNode returns pods that are nominated to run on the given node,
+// but they are waiting for other pods to be removed from the node before they
+// can be actually scheduled.
+func (npm *nominatedPodMap) NominatedPodsForNode(nodeName string) []*v1.Pod {
+	npm.RLock()
+	defer npm.RUnlock()
+	if list, ok := npm.nominatedPods[nodeName]; ok {
+		return list
+	}
+	return nil
```

Comment:
do you need the check, or could you just `return npm.nominatedPods[nodeName]`?

Reply:
Just return `npm.nominatedPods[nodeName]` is fine.